server/services/store/sqlstore/initialize.go

```diff
@@ -28,60 +49,88 @@ func (s *SQLStore) InitializeTemplates() error {
 // ToDo: needs a rewrite to loop over teams and insert boards and
 // blocks, commented to move forward with the rest of the store
 // changes
-func (s *SQLStore) importInitialTemplates() error {
-	/*
-		s.logger.Debug("importInitialTemplates")
-		blocksJSON := initializations.MustAsset("templates.json")
-
-		var archive model.Archive
-		err := json.Unmarshal(blocksJSON, &archive)
-		if err != nil {
-			return err
-		}
+func (s *SQLStore) importInitialTemplates(db sq.BaseRunner, teamIDs []string) error {
+	s.logger.Debug("importInitialTemplates")
+	blocksJSON := initializations.MustAsset("templates.json")
 
-		// ToDo: loop over teams
+	var archive model.Archive
+	err := json.Unmarshal(blocksJSON, &archive)
+	if err != nil {
+		return err
+	}
 
-		globalContainer := store.Container{
-			TeamID: "0",
-		}
+	for _, teamID := range teamIDs {
+		for _, board := range archive.Boards {
+			boardBlocks := []model.Block{}
+			newBoardID := utils.NewID(utils.IDTypeBoard)
+			for _, block := range archive.Blocks {
+				if block.ParentID == board.ID || block.RootID == board.ID {
+					boardBlocks = append(boardBlocks, block)
+				}
+			}
 
-		s.logger.Debug("Inserting blocks", mlog.Int("block_count", len(archive.Blocks)))
-		for i := range archive.Blocks {
-			s.logger.Trace("insert block",
-				mlog.String("blockID", archive.Blocks[i].ID),
-				mlog.String("block_type", archive.Blocks[i].Type.String()),
-				mlog.String("block_title", archive.Blocks[i].Title),
-			)
-			err := s.InsertBlock(globalContainer, &archive.Blocks[i], "system")
+			boardBlocks = model.GenerateBlockIDs(boardBlocks)
+			updatedBoardBlocks := []model.Block{}
+
+			for _, block := range boardBlocks {
+				updatedBlock := block
+				if updatedBlock.ParentID == updatedBlock.ID {
+					updatedBlock.ParentID = newBoardID
+				}
+				if updatedBlock.RootID == updatedBlock.ID {
+					updatedBlock.RootID = newBoardID
+				}
+				updatedBlock.BoardID = newBoardID
+				updatedBoardBlocks = append(updatedBoardBlocks, updatedBlock)
+			}
+			board.ID = newBoardID
+			board.Type = model.BoardTypeOpen
+			board.TeamID = teamID
+			board.IsTemplate = true
+
+			for _, block := range updatedBoardBlocks {
+				err := s.insertBlock(db, &block, "system")
```

Comment:
I get a `gosec` memory aliasing error here in the IDE. Inlining the `insertBlock` in error check fixes it.

<img width="706" alt="Screenshot 2022-01-12 at 5 40 42 PM" src="https://user-images.githubusercontent.com/18575143/149137918-0077d066-bece-4f1c-a645-ccfd3cac7f84.png">